ruby/ext/google/protobuf_c/message.c

```diff
@@ -329,6 +329,30 @@ VALUE Message_inspect(VALUE _self) {
   return str;
 }
 
+
+VALUE Message_to_h(VALUE _self) {
+  MessageHeader* self;
+  TypedData_Get_Struct(_self, MessageHeader, &Message_type, self);
+
+  VALUE hash = rb_hash_new();
+  rb_gc_register_address(&hash);
```

Comment:
Oh, definitely not! `rb_gc_register_address` registers a static root with the garbage collector -- e.g., for a global variable. You're telling the GC that some random address on this thread's stack is a root to be kept alive. Probably not what you want :-)